docs/designs/045-node-validation.md

```
@@ -0,0 +1,445 @@
+* **Support for validation sessions**: a quarantine session in NVSentinel is made up of a group of faults all of which must be remediated prior to the node being unquarantined. A similar capability is provided with the validation-controller where all validation requests against a given node must be satisfied prior to the node being uncordoned.
+    * For example, if a new node requires validation and experiences a fault that results in NVSentinel also requesting validation, both sets of validation tests must succeed prior to the node being uncordoned.
+* **Common validation interface:** a shared interface for requesting validation prevents all clients from needing to maintain logic to work with the different test providers, including our supported K8s job and NCCL test providers. A client who wants to validate a group of nodes or GPUs only needs to know the test name and we will maintain a common entrypoint in the validation-controller.
+* **Batching multiple validation requests:** if a single client independently creates validation requests across nodes or multiple clients request validation, a dedicated controller provides the opportunity to batch multiple validation requests together. This allows for de-duplicating validation requests across the same node for the same test and also batching nodes together in multi-node tests. This prevents clients from needing to handle batching on their side prior to issuing validation requests.
+    * For example, if NVSentinel remediates 2 nodes and requests that both undergo a multi-node NCCL test, we have the opportunity to include those 2 nodes together in a single test from our NCCL provider.
+* **Batching multiple nodes into a single validation request:** a validation request will allow targeting a single node or a group of nodes (along with a single GPU or all GPUs on a single node). If a test provider only supports targeting a single node, such as the K8s job provider, the validation-controller will handle creating a validation job per node without requiring the requesting client to create and track the execution of each job. Clients will only need to monitor the overall status for its validation request.
+
+## Validation API and Configuration
+
+In order to run validation tests against GPU nodes, a cluster operator will need the ability to define a set of supported validation tests. Additionally, a client requesting that a GPU node undergoes validation will need to provide the following inputs:
+
+```
```

markdownlint-cli2 (0.23.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)